_posts/frameworks/2016-12-19-spring-restdocs.md

```diff
@@ -0,0 +1,274 @@
+    "self" : {
+      "href" : "http://localhost:8080/projects/1"
+    },
+    "files" : {
+      "href" : "http://localhost:8080/projects/1/files"
+    },
+    "analyzers" : {
+      "href" : "http://localhost:8080/projects/1/analyzers"
+    },
+    "strategy" : {
+      "href" : "http://localhost:8080/projects/1/strategy"
+    }
+  }
+}
+----
+```
+
+These examples already go a long way to documenting your REST API. Examples are the best way
+for developers to get to know your API. The snippets automatically generated from your test
+don't help when they rot in your snippets folder, though, so we have to expose them by 
+including them into a central documentation of some sorts.
+
+## Creating API Docs with Asciidoctor
+With the snippets at hand, we can now create our API documentation. The snippets are in 
+[Asciidoctor](http://asciidoctor.org/) format by default. Asciidoctor is a markup language 
+similiar to Markdown, but much more powerful. You can now simply create an Asciidoctor 
+document with your favorite text editor. That document will provide the stage for including the 
+snippets. An example document would look like this:
+
+```asciidoctor
+= My REST API
+v{version}, Tom Hombergs, {date}
+:doctype: book
+:icons: font
+:source-highlighter: highlightjs
+:highlightjs-theme: github
+:toc: left
+:toclevels: 3
+:sectlinks:
+:sectnums:
+
+[introduction]
+== Introduction
+... some warm introductory words... .
+
+== Creating a Project
+
+=== Example Request
+include::{snippets}/projects/create/http-request.adoc[]
+
+=== Example Response
+include::{snippets}/projects/create/http-response.adoc[]
+
+```
+
+The document above includes the example HTTP request and response snippets that are generated by
+the integration test above. While it could yet be fleshed 
+out with a little more text, the documentation above is already worth its weight in gold 
+(imagine each byte weighing a pound or so...). Even if you change
+your implementation, you will not have to touch your documentation, since the example snippets 
+will be generated fresh with each build and thus be up-to-date at all times! You still have to 
+include the generation of your snippets into your build though, which we will have a look at
+in the next section
+
+## Integrating Documentation into your Build
+The integration tests should run with each build. Thus, our documentation snippets are generated
+with each build. The missing step now is to generate human-readable documentation from your
+asciidoctor document.
+
+This can be done using the [Asciidoctor Gradle Plugin](http://asciidoctor.org/docs/asciidoctor-gradle-plugin/)
+when you're using Gradle as your build tool or the [Asciidoctor Maven Plugin](http://asciidoctor.org/docs/asciidoctor-maven-plugin/)
+when you're using Maven. The following examples are based on Gradle.
+
+In your `build.gradle`, you will first have to define a dependency to the plugin:
+
+```groovy
+buildscript {
+    repositories {
+        mavenCentral()
```

Comment:
just use `jcenter()` as `mavenCentral()` is included in `jcenter()` and we don't need `mavenLocal()`, right?